server/app/models/grid.rb

```diff
@@ -83,11 +92,10 @@ def create_node!(name, ensure_unique_name: false, **attrs)
 
       if self.host_nodes.where(node_number: node.node_number).exists?
         warn "retry node #{name} node_number allocation on error: #{exc}"
-        node.name = name
         retry
       elsif ensure_unique_name && self.host_nodes.where(name: node.name).exists?
         warn "rename node #{name} on name conflict: #{exc}"
-        node.name = "#{name}-#{node.node_number}"
```

Comment:
This logic was broken, it would re-use the wrong `node_number` for the suffix if that node_number happened to be allocated for a different node while this was retrying.